[CI] AzureRepositoryAnalysisRestIT testRepositoryAnalysis failing

**Build Scans:**
- [elasticsearch-pull-request #137563 / part-2](https://gradle-enterprise.elastic.co/s/gignayq3b5wzg)
- [elasticsearch-intake #37622 / part2](https://gradle-enterprise.elastic.co/s/4u3d6rmbx7l2e)
- [elasticsearch-pull-request #137536 / part-2](https://gradle-enterprise.elastic.co/s/qjksnynx2dtqg)
- [elasticsearch-pull-request #137461 / part-2](https://gradle-enterprise.elastic.co/s/rptm3ujajspd2)
- [elasticsearch-pull-request #137461 / part-2](https://gradle-enterprise.elastic.co/s/ayrbviv4tcc2a)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:snapshot-repo-test-kit:qa:azure:javaRestTest" --tests "org.elasticsearch.repositories.blobstore.testkit.analyze.AzureRepositoryAnalysisRestIT.testRepositoryAnalysis" -Dtests.seed=BE5529B2BF0E6582 -Dtests.locale=es-NI -Dtests.timezone=Europe/Isle_of_Man -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.repositories.blobstore.testkit.analyze.AzureRepositoryAnalysisRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRepositoryAnalysis'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Uncaught exception
```

**Issue Reasons:**
- [main] 1 consecutive failures in test testRepositoryAnalysis
- [main] 1 consecutive failures in step part-2
- [main] 1 consecutive failures in pipeline elasticsearch-pull-request
- [main] 5 failures in test testRepositoryAnalysis (0.6% fail rate in 832 executions)
- [main] 4 failures in step part-2 (2.1% fail rate in 194 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (1.6% fail rate in 189 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -301,9 +301,6 @@ tests:
 - class: org.elasticsearch.index.reindex.ReindexClientYamlTestSuiteIT
   method: test {yaml=reindex/10_basic/Response format for version conflict with conflicts=proceed}
   issue: https://github.com/elastic/elasticsearch/issues/146419
-- class: org.elasticsearch.repositories.blobstore.testkit.analyze.AzureRepositoryAnalysisRestIT
-  method: testRepositoryAnalysis
-  issue: https://github.com/elastic/elasticsearch/issues/146430
 - class: org.elasticsearch.xpack.esql.optimizer.rules.physical.local.SubstituteRoundToGoldenTests
   method: testDateTruncNotTransformToQueryAndTags
   issue: https://github.com/elastic/elasticsearch/issues/146458

--- a/test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpFixture.java
+++ b/test/fixtures/azure-fixture/src/main/java/fixture/azure/AzureHttpFixture.java
@@ -14,10 +14,15 @@ import com.sun.net.httpserver.HttpsServer;
 
 import org.elasticsearch.common.Strings;
 import org.elasticsearch.common.network.InetAddresses;
+import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.ssl.KeyStoreUtil;
 import org.elasticsearch.common.ssl.PemUtils;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.common.util.concurrent.ThreadContext;
 import org.elasticsearch.core.Nullable;
 import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.TestEsExecutors;
+import org.elasticsearch.threadpool.ThreadPool;
 import org.junit.rules.ExternalResource;
 
 import java.io.IOException;
@@ -29,6 +34,8 @@ import java.security.SecureRandom;
 import java.security.cert.Certificate;
 import java.util.List;
 import java.util.Objects;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
 import java.util.function.Predicate;
 
 import javax.net.ssl.KeyManager;
@@ -51,6 +58,7 @@ public class AzureHttpFixture extends ExternalResource {
     private HttpServer server;
     private HttpServer metadataServer;
     private HttpServer oauthTokenServiceServer;
+    private ExecutorService executorService;
 
     // JWT-looking value for workload identity authentication -- the exact value has no inherent meaning
     private final String federatedToken = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9";
@@ -200,6 +208,16 @@ public class AzureHttpFixture extends ExternalResource {
                 this.metadataServer = HttpServer.create(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0), 0);
                 metadataServer.createContext("/", new AzureMetadataServiceHttpHandler(managedIdentityBearerToken));
                 metadataServer.start();
+                this.executorService = EsExecutors.newScaling(
+                    "azure-http-fixture",
+                    1,
+                    100,
+                    30,
+                    TimeUnit.SECONDS,
+                    true,
+                    TestEsExecutors.testOnlyDaemonThreadFactory("s3-http-fixture"),
+                    new ThreadContext(Settings.EMPTY)
+                );
             }
 
             switch (protocol) {
@@ -211,6 +229,7 @@ public class AzureHttpFixture extends ExternalResource {
                         "/" + account,
                         new AzureHttpHandler(account, container, actualAuthHeaderPredicate, leaseExpiryPredicate)
                     );
+                    server.setExecutor(executorService);
                     server.start();
 
                     oauthTokenServiceServer = HttpServer.create(new InetSocketAddress(InetAddress.getLoopbackAddress(), 0), 0);
@@ -243,6 +262,7 @@ public class AzureHttpFixture extends ExternalResource {
                             "/" + account,
                             new AzureHttpHandler(account, container, actualAuthHeaderPredicate, leaseExpiryPredicate)
                         );
+                        httpsServer.setExecutor(executorService);
                         httpsServer.start();
                     }
                     {
@@ -253,6 +273,7 @@ public class AzureHttpFixture extends ExternalResource {
                             "/",
                             new AzureOAuthTokenServiceHttpHandler(workloadIdentityBearerToken, federatedToken, tenantId, clientId)
                         );
+                        httpsServer.setExecutor(executorService);
                         httpsServer.start();
                     }
                 }
@@ -281,6 +302,7 @@ public class AzureHttpFixture extends ExternalResource {
             server.stop(0);
             metadataServer.stop(0);
             oauthTokenServiceServer.stop(0);
+            ThreadPool.terminate(executorService, 10, TimeUnit.SECONDS);
         }
     }
 }

